executor/include/execution/property/base_property.hpp

```diff
@@ -4,25 +4,25 @@
 
 #pragma once
 
+#include <execution/type_trait.hpp>
 #include <functional>
 #include <iostream>
 #include <string>
 
-#include <execution/type_trait.hpp>
-
 template <typename Derived, bool requireable, bool preferable>
 class basic_executor_property {
  public:
   static constexpr bool is_requirable = requireable;
   static constexpr bool is_preferable = preferable;
 
-  template <class T> static constexpr bool is_applicable_property() {
+  template <class T>
+  static constexpr bool is_applicable_property() {
     return execution::is_executor<T>();
   }
 
   template <class Executor>
   static constexpr auto static_query()
-  -> decltype(Executor::query(std::declval<Derived>())) {
+      -> decltype(Executor::query(std::declval<Derived>())) {
```

Comment:
All query methods will always return bool right? No need to complicate.

Reply:
Have you considered defining this method outside the class as a free-form function with two template parameters. renamed simply `query`?

Reply:
> All query methods will always return bool right?
Actually no, if you query the property `ProtoAllocator` it should return the allocator itself. I haven't implemented it currently but it is to be done.

> defining this method outside the class 
Like `execution::query()`? As that is already defined in `can_query.hpp`, currently, the code offers query both as a member function and free function. I will look into the proposal and see what they have done.

Reply:
> > All query methods will always return bool right?
> 
> Actually no, if you query the property `ProtoAllocator` it should return the allocator itself. I haven't implemented it currently but it is to be done.

Isn't that inconsistent with the rest of your query functions? All of them return `bool` except one. What's the reason for this different behavior?

> > defining this method outside the class
> 
> Like `execution::query()`? As that is already defined in `can_query.hpp`, currently, the code offers query both as a member function and free function.

I'm ok with the first two
```cpp
query<Executor, Property>(); //ok
executor.query<property>(); //ok
Property::static_query<Executor>(); // This forces you to have CRTP
```
You're imposing CRTP into your class design just to offer that method, and that's the trade-off I don't agree

Reply:
Let me revaluate why the TS specifies that and come up with a  convincing argument for its need.